### ISSUE TITLE
Splunk App Connection with Wazuh API

### ISSUE THREAD

**Issue by DMHS** [wazuh/wazuh-splunk#40]
I have installed the Splunk app and setup the index and universal forwarder according to the documentation. Splunk is indexing data from the Wazuh Manager instance and that is populating 'Top Alerts' overview tab in the Splunk App.

When I enter the Wazuh API credentials into the Global Configuration tab of the Splunk App no data is populated in the 'Manager Status' page nor the 'Agents Summary' page.

I can curl the 55000 Wazuh API port from the Splunk instance. What am I missing here? Any help is appreciated. Thank you.

**Comment by manuasir** [comment]
Hello @DMHS ,

Could you please show up here a snapshot of your Global Configuration view? Maybe the 'base ip' and 'base port' fields weren't filled correctly with the IP and port of the machine where Splunk Enterprise is running. 

Regards

**Comment by DMHS** [comment]
Hi manuasir,

Here is a screenshot of the Global Configuration view. I have changed the username and password to default for purpose of the screenshot.

![image](https://user-images.githubusercontent.com/22699469/39056739-23687b02-447d-11e8-9970-5e2af2a87c3e.png)

Also, here is a screenshot of a succesful curl from the Splunk box to the Wazuh box.
![image](https://user-images.githubusercontent.com/22699469/39057092-0c6e33be-447e-11e8-8d7a-0e003a8bf251.png)

Thank you.

**Comment by manuasir** [comment]
Hi @DMHS ,

Everything seems to be correct in Global configuration, so let's check a few different things.
For knowing the context better, could you please show up here your Wazuh App for Splunk version? You can get it by listing the 'app.conf' file:

`cat $SPLUNK_HOME/etc/apps/wazuh/default/app.conf`

Also, let's check the App backend connectivity. Please, perform a **curl** to this endpoint example as following:

`http://<BASE-IP>:<BASE-PORT>/en-US/custom/wazuh/manager/status?ip=<API-IP>&port=<API-PORT>&user=<API-USER>&pass=<API-PASS>`

Try that **curl** command with 'http' and also with 'https' and please show up the results here.

Aditionally, it would be really helpful if you can navigate to any tab that use Wazuh API, such Manager->Manager status  with the development console opened in your browser, it's usually opened with F12 key. Please, post here any error in the console too.

Thank you for your patience,
Kind regards

**Comment by DMHS** [comment]
app.conf
![image](https://user-images.githubusercontent.com/22699469/39061592-6efc052c-448a-11e8-9144-9ed78b909f54.png)

curl command with http
![image](https://user-images.githubusercontent.com/22699469/39061670-b9fc0c7a-448a-11e8-8633-fd75d65c56aa.png)

curl command with https
![image](https://user-images.githubusercontent.com/22699469/39062274-af78b3b4-448c-11e8-8766-d00744a8d691.png)
![image](https://user-images.githubusercontent.com/22699469/39062306-c345c904-448c-11e8-84c4-fafc4d11e248.png)

Manager Status
![image](https://user-images.githubusercontent.com/22699469/39062764-2941989a-448e-11e8-935e-967e62f29818.png)

**Comment by DMHS** [comment]
I have a feeling that it has to do with my Splunk instance using HTTPS. Also, I am using a dev license if that matters.

**Comment by manuasir** [comment]
@DMHS Sorry, I forgot to mention that please use the curl argument between double quotes like this:
```
curl "http://<BASE-IP>:<BASE-PORT>/en-US/custom/wazuh/manager/status?ip=<API-IP>&port=<API-PORT>&user=<API-USER>&pass=<API-PASS>"
```
Otherwise it fails. 
Kind regargs

**Comment by DMHS** [comment]
Ok, sorry about that.

curl with http
![image](https://user-images.githubusercontent.com/22699469/39063899-1b002da6-4492-11e8-8c5a-e0a43a38a1be.png)

curl with https
![image](https://user-images.githubusercontent.com/22699469/39063968-5a591634-4492-11e8-87fa-825d46e958cb.png)

**Comment by DMHS** [comment]
Sounds good. Thank you very much for your help.

**Comment by manuasir** [comment]
Hello @DMHS ,

As you see I marked this issue as closed since it's fixed in the last v2.2.0 release. You can follow these steps in order to upgrade to the new app version:
- Stop the Splunk service:
```
$SPLUNK_HOME/bin/splunk stop
```
- Download the source code of last Wazuh app for Splunk release: [v2.2.0](https://github.com/wazuh/wazuh-splunk/releases/download/v2.2.0/wazuh-3.2.2-2.2.0.zip) 
- Delete old Wazuh app for Splunk folder:
```
rm -rf $SPLUNK_HOME/etc/apps/wazuh/
```
- Unzip the last version into Splunk apps folder:
```
unzip wazuh-3.2.2-2.2.0.zip -d $SPLUNK_HOME/etc/apps/
```
- Start Splunk service:
```
$SPLUNK_HOME/bin/splunk start
```
Notice that any index will be deleted or modify along this process, also remember that **$SPLUNK_HOME is a reference to your Splunk base directory, by default /opt/splunk/**. Change it in accordance with your installation if changed.
I'd like to thank you for your report and please, do not hesitate to open another issue if needed.

Regards,
Manu
